esphome/components/nau7802/nau7802.cpp

```diff
@@ -0,0 +1,285 @@
+      voltage = "3.0V";
+      break;
+    case NAU7802_LDO_3V3:
+      voltage = "3.3V";
+      break;
+    case NAU7802_LDO_3V6:
+      voltage = "3.6V";
+      break;
+    case NAU7802_LDO_3V9:
+      voltage = "3.9V";
+      break;
+    case NAU7802_LDO_4V2:
+      voltage = "4.2V";
+      break;
+    case NAU7802_LDO_4V5:
+      voltage = "4.5V";
+      break;
+    case NAU7802_LDO_EXTERNAL:
+      voltage = "External";
+      break;
+  }
+  ESP_LOGCONFIG(TAG, "  LDO Voltage: %s", voltage.c_str());
+  int gain = 0;
+  switch (this->gain_) {
+    case NAU7802_GAIN_128:
+      gain = 128;
+      break;
+    case NAU7802_GAIN_64:
+      gain = 64;
+      break;
+    case NAU7802_GAIN_32:
+      gain = 32;
+      break;
+    case NAU7802_GAIN_16:
+      gain = 16;
+      break;
+    case NAU7802_GAIN_8:
+      gain = 8;
+      break;
+    case NAU7802_GAIN_4:
+      gain = 4;
+      break;
+    case NAU7802_GAIN_2:
+      gain = 2;
+      break;
+    case NAU7802_GAIN_1:
+      gain = 1;
+      break;
+  }
+  ESP_LOGCONFIG(TAG, "  Gain: %dx", gain);
+  int sps = 0;
+  switch (this->sps_) {
+    case NAU7802_SPS_320:
+      sps = 320;
+      break;
+    case NAU7802_SPS_80:
+      sps = 80;
+      break;
+    case NAU7802_SPS_40:
+      sps = 40;
+      break;
+    case NAU7802_SPS_20:
+      sps = 20;
+      break;
+    case NAU7802_SPS_10:
+      sps = 10;
+      break;
+  }
+  ESP_LOGCONFIG(TAG, "  Samples Per Second: %d", sps);
+  LOG_UPDATE_INTERVAL(this);
+}
+
+void NAU7802Sensor::write_value_(uint8_t start_reg, size_t size, int32_t value) {
+  uint8_t data[4];
+  for (int i = 0; i < size; i++) {
+    data[i] = 0xFF & (value >> (size - 1 - i) * 8);
+  }
+  this->write_register(start_reg, data, size);
+}
+
+int32_t NAU7802Sensor::read_value_(uint8_t start_reg, size_t size) {
+  uint8_t data[4];
+  this->read_register(start_reg, data, size);
+  int32_t result = 0;
+  for (int i = 0; i < size; i++) {
+    result |= data[i] << (size - 1 - i) * 8;
+  }
+  // extend sign bit
+  if (result & 0x800000 && size == 3) {
+    result |= 0xFF000000;
+  }
+  return result;
+}
+
+bool NAU7802Sensor::calibrate_(enum NAU7802CalibrationModes mode) {
+  i2c::I2CRegister ctrl2 = this->reg(CTRL2_REG);
+  // Calibrate
+  ctrl2 |= mode;
+  ctrl2 |= CTRL2_CALS;
+  while (ctrl2.get() & CTRL2_CALS) {
+    delay(10);
+  }
+
+  if (ctrl2.get() & CTRL2_CAL_ERR) {
+    ESP_LOGE(TAG, "Failed to calibrate sensor");
+    this->mark_failed();
+    return false;
+  }
+  int32_t ocal = this->read_value_(OCAL1_B2_REG, 3);
+  ESP_LOGI(TAG, "New Offset: %s", to_string(ocal).c_str());
+  uint32_t gcal = this->read_value_(GCAL1_B3_REG, 4);
+  float gcal_f = ((float) gcal / (float) (1 << GCAL1_FRACTIONAL));
+  ESP_LOGI(TAG, "New Gain: %f", gcal_f);
+  return true;
+}
+
+float NAU7802Sensor::get_setup_priority() const { return setup_priority::DATA; }
+
+void NAU7802Sensor::update() {
+  if (this->is_failed()) {
+    return;
+  }
+
+  if (!is_data_ready_()) {
```

Comment:
> All uses of class members and member functions should be prefixed with `this->` to distinguish them from global functions in code review.
https://esphome.io/guides/contributing#codebase-standards

```suggestion
  if (!this->is_data_ready_()) {
```